scripts/assets/file_content_validation_checks.textproto

```diff
@@ -306,3 +306,50 @@ file_content_checks {
   exempted_file_name: "utility/src/test/java/org/oppia/android/util/math/RealExtensionsTest.kt"
   exempted_file_patterns: "testing/src/main/java/org/oppia/android/testing/junit/.+?\\.kt"
 }
+
+file_content_checks {
+  file_path_regex: "app/src/main/res/values/component_colors.xml|app/src/main/res/values/color_palette.xml"
+  prohibited_content_regex: "name=.+(?<!_color).>@|name=.*[A-Z].*>@"
+  failure_message: "All color declarations in component_color.xml and color_palette.xml should end with _color suffix following snake_case naming convention."
+}
+file_content_checks {
+  file_path_regex: "app/src/main/res/values/color_defs.xml"
+  prohibited_content_regex: "name=.+(?<=color).>#|name=.*[A-Z].*>#"
+  failure_message: "All color declarations in color_defs.xml should be named using snake_case convention and not contain color keyword."
+}
+file_content_checks {
```

Comment:
Does the migration file also need to be added as an exemption, or is the plan to check this in after the migration is complete?

Reply:
I was thinking to not consider migration file for tests at this moment as we are not expecting to add anything to it anyway. Can you elaborate more on what you mean?

Reply:
Ah, well the main consideration I'm having is whether this PR should also be introducing enforcement of component_colors in layout files (as that's the missing link--at the moment, layout files are allowed to reference anything). My original question came from me thinking about the fact that we need to be isolating references of each type of color, but that actually happens by default with the way you've approached it so long as layouts and Kotlin/Java code is limited to only referencing component colors.

Is the plan to introduce that support later on?

Reply:
That sounds fine to me, but does that mean #4203 isn't being fully resolved, or should we split the later work into a follow-up issue? I just want to make sure that nothing is lost here given that the later checks are sort of critical for getting the most value out of the checks introduced in this PR.

Reply:
Replied on the main thread.